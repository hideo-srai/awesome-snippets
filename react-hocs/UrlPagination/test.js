// @flow
import { start } from '../../testing'

describe('withPagination', () => {
  it('renders examples', () => start().examples())

  it('default example', () => start()
    .example('Advanced', { capture: false })
    .press('page-change-button')
    .assertUrlQueryParam('advancedTable_page', '2')
  )
})
