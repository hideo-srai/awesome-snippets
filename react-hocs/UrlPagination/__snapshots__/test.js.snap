// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withPagination renders examples: lg 1`] = `
Array [
  <div>
    <div>
      This table uses withPaginationHoc!
    </div>
    <div>
      Item indexes to show:
       
      0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19
    </div>
    <div>
      Page:
       
      0
    </div>
    <div>
      Per page:
       
      20
    </div>
    <button
      onClick={[Function]}
    >
      Random page
    </button>
    <button
      onClick={[Function]}
    >
      Change Per Page
    </button>
  </div>,
  <div>
    <div>
      Showing: 
      1
      -
      4
      /
      7
      , Sort By: 
      , Sort Direction: 
      desc
    </div>
    <button
      data-test-id="page-change-button"
      onClick={[Function]}
    >
      Next Page
    </button>
    <button
      onClick={[Function]}
    >
      Per Page
    </button>
    <button
      onClick={[Function]}
    >
      SortBy
    </button>
    <button
      onClick={[Function]}
    >
      SortDirection
    </button>
    <table
      width="100%"
    >
      <tbody>
        <tr>
          <td>
            25
          </td>
          <td>
            John Doe
          </td>
          <td>
            djohn@vydia.test
          </td>
        </tr>
        <tr>
          <td>
            40
          </td>
          <td>
            Lucas Leon
          </td>
          <td>
            lucas.leon@vydia.test
          </td>
        </tr>
        <tr>
          <td>
            30
          </td>
          <td>
            Hideo Shirai
          </td>
          <td>
            shirai.h@vydia.test
          </td>
        </tr>
        <tr>
          <td>
            28
          </td>
          <td>
            Donald Smith
          </td>
          <td>
            xman.donald@vydia.test
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <div>
    <div>
      Item indexes to show:
       
      0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19
    </div>
    <div>
      Page:
       
      0
    </div>
    <div>
      Per page:
       
      20
    </div>
    <button
      onClick={[Function]}
    >
      Random page
    </button>
    <button
      onClick={[Function]}
    >
      Change Per Page
    </button>
    <hr />
    <div>
      Item indexes to show:
       
      0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19
    </div>
    <div>
      Page:
       
      0
    </div>
    <div>
      Per page:
       
      20
    </div>
    <button
      onClick={[Function]}
    >
      Random page
    </button>
    <button
      onClick={[Function]}
    >
      Change Per Page
    </button>
  </div>,
]
`;

exports[`withPagination renders examples: md 1`] = `
Array [
  <div>
    <div>
      This table uses withPaginationHoc!
    </div>
    <div>
      Item indexes to show:
       
      0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19
    </div>
    <div>
      Page:
       
      0
    </div>
    <div>
      Per page:
       
      20
    </div>
    <button
      onClick={[Function]}
    >
      Random page
    </button>
    <button
      onClick={[Function]}
    >
      Change Per Page
    </button>
  </div>,
  <div>
    <div>
      Showing: 
      1
      -
      4
      /
      7
      , Sort By: 
      , Sort Direction: 
      desc
    </div>
    <button
      data-test-id="page-change-button"
      onClick={[Function]}
    >
      Next Page
    </button>
    <button
      onClick={[Function]}
    >
      Per Page
    </button>
    <button
      onClick={[Function]}
    >
      SortBy
    </button>
    <button
      onClick={[Function]}
    >
      SortDirection
    </button>
    <table
      width="100%"
    >
      <tbody>
        <tr>
          <td>
            25
          </td>
          <td>
            John Doe
          </td>
          <td>
            djohn@vydia.test
          </td>
        </tr>
        <tr>
          <td>
            40
          </td>
          <td>
            Lucas Leon
          </td>
          <td>
            lucas.leon@vydia.test
          </td>
        </tr>
        <tr>
          <td>
            30
          </td>
          <td>
            Hideo Shirai
          </td>
          <td>
            shirai.h@vydia.test
          </td>
        </tr>
        <tr>
          <td>
            28
          </td>
          <td>
            Donald Smith
          </td>
          <td>
            xman.donald@vydia.test
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <div>
    <div>
      Item indexes to show:
       
      0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19
    </div>
    <div>
      Page:
       
      0
    </div>
    <div>
      Per page:
       
      20
    </div>
    <button
      onClick={[Function]}
    >
      Random page
    </button>
    <button
      onClick={[Function]}
    >
      Change Per Page
    </button>
    <hr />
    <div>
      Item indexes to show:
       
      0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19
    </div>
    <div>
      Page:
       
      0
    </div>
    <div>
      Per page:
       
      20
    </div>
    <button
      onClick={[Function]}
    >
      Random page
    </button>
    <button
      onClick={[Function]}
    >
      Change Per Page
    </button>
  </div>,
]
`;

exports[`withPagination renders examples: sm 1`] = `
Array [
  <div>
    <div>
      This table uses withPaginationHoc!
    </div>
    <div>
      Item indexes to show:
       
      0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19
    </div>
    <div>
      Page:
       
      0
    </div>
    <div>
      Per page:
       
      20
    </div>
    <button
      onClick={[Function]}
    >
      Random page
    </button>
    <button
      onClick={[Function]}
    >
      Change Per Page
    </button>
  </div>,
  <div>
    <div>
      Showing: 
      1
      -
      4
      /
      7
      , Sort By: 
      , Sort Direction: 
      desc
    </div>
    <button
      data-test-id="page-change-button"
      onClick={[Function]}
    >
      Next Page
    </button>
    <button
      onClick={[Function]}
    >
      Per Page
    </button>
    <button
      onClick={[Function]}
    >
      SortBy
    </button>
    <button
      onClick={[Function]}
    >
      SortDirection
    </button>
    <table
      width="100%"
    >
      <tbody>
        <tr>
          <td>
            25
          </td>
          <td>
            John Doe
          </td>
          <td>
            djohn@vydia.test
          </td>
        </tr>
        <tr>
          <td>
            40
          </td>
          <td>
            Lucas Leon
          </td>
          <td>
            lucas.leon@vydia.test
          </td>
        </tr>
        <tr>
          <td>
            30
          </td>
          <td>
            Hideo Shirai
          </td>
          <td>
            shirai.h@vydia.test
          </td>
        </tr>
        <tr>
          <td>
            28
          </td>
          <td>
            Donald Smith
          </td>
          <td>
            xman.donald@vydia.test
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <div>
    <div>
      Item indexes to show:
       
      0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19
    </div>
    <div>
      Page:
       
      0
    </div>
    <div>
      Per page:
       
      20
    </div>
    <button
      onClick={[Function]}
    >
      Random page
    </button>
    <button
      onClick={[Function]}
    >
      Change Per Page
    </button>
    <hr />
    <div>
      Item indexes to show:
       
      0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19
    </div>
    <div>
      Page:
       
      0
    </div>
    <div>
      Per page:
       
      20
    </div>
    <button
      onClick={[Function]}
    >
      Random page
    </button>
    <button
      onClick={[Function]}
    >
      Change Per Page
    </button>
  </div>,
]
`;

exports[`withPagination renders examples: xl 1`] = `
Array [
  <div>
    <div>
      This table uses withPaginationHoc!
    </div>
    <div>
      Item indexes to show:
       
      0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19
    </div>
    <div>
      Page:
       
      0
    </div>
    <div>
      Per page:
       
      20
    </div>
    <button
      onClick={[Function]}
    >
      Random page
    </button>
    <button
      onClick={[Function]}
    >
      Change Per Page
    </button>
  </div>,
  <div>
    <div>
      Showing: 
      1
      -
      4
      /
      7
      , Sort By: 
      , Sort Direction: 
      desc
    </div>
    <button
      data-test-id="page-change-button"
      onClick={[Function]}
    >
      Next Page
    </button>
    <button
      onClick={[Function]}
    >
      Per Page
    </button>
    <button
      onClick={[Function]}
    >
      SortBy
    </button>
    <button
      onClick={[Function]}
    >
      SortDirection
    </button>
    <table
      width="100%"
    >
      <tbody>
        <tr>
          <td>
            25
          </td>
          <td>
            John Doe
          </td>
          <td>
            djohn@vydia.test
          </td>
        </tr>
        <tr>
          <td>
            40
          </td>
          <td>
            Lucas Leon
          </td>
          <td>
            lucas.leon@vydia.test
          </td>
        </tr>
        <tr>
          <td>
            30
          </td>
          <td>
            Hideo Shirai
          </td>
          <td>
            shirai.h@vydia.test
          </td>
        </tr>
        <tr>
          <td>
            28
          </td>
          <td>
            Donald Smith
          </td>
          <td>
            xman.donald@vydia.test
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <div>
    <div>
      Item indexes to show:
       
      0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19
    </div>
    <div>
      Page:
       
      0
    </div>
    <div>
      Per page:
       
      20
    </div>
    <button
      onClick={[Function]}
    >
      Random page
    </button>
    <button
      onClick={[Function]}
    >
      Change Per Page
    </button>
    <hr />
    <div>
      Item indexes to show:
       
      0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19
    </div>
    <div>
      Page:
       
      0
    </div>
    <div>
      Per page:
       
      20
    </div>
    <button
      onClick={[Function]}
    >
      Random page
    </button>
    <button
      onClick={[Function]}
    >
      Change Per Page
    </button>
  </div>,
]
`;

exports[`withPagination renders examples: xs 1`] = `
Array [
  <div>
    <div>
      This table uses withPaginationHoc!
    </div>
    <div>
      Item indexes to show:
       
      0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19
    </div>
    <div>
      Page:
       
      0
    </div>
    <div>
      Per page:
       
      20
    </div>
    <button
      onClick={[Function]}
    >
      Random page
    </button>
    <button
      onClick={[Function]}
    >
      Change Per Page
    </button>
  </div>,
  <div>
    <div>
      Showing: 
      1
      -
      4
      /
      7
      , Sort By: 
      , Sort Direction: 
      desc
    </div>
    <button
      data-test-id="page-change-button"
      onClick={[Function]}
    >
      Next Page
    </button>
    <button
      onClick={[Function]}
    >
      Per Page
    </button>
    <button
      onClick={[Function]}
    >
      SortBy
    </button>
    <button
      onClick={[Function]}
    >
      SortDirection
    </button>
    <table
      width="100%"
    >
      <tbody>
        <tr>
          <td>
            25
          </td>
          <td>
            John Doe
          </td>
          <td>
            djohn@vydia.test
          </td>
        </tr>
        <tr>
          <td>
            40
          </td>
          <td>
            Lucas Leon
          </td>
          <td>
            lucas.leon@vydia.test
          </td>
        </tr>
        <tr>
          <td>
            30
          </td>
          <td>
            Hideo Shirai
          </td>
          <td>
            shirai.h@vydia.test
          </td>
        </tr>
        <tr>
          <td>
            28
          </td>
          <td>
            Donald Smith
          </td>
          <td>
            xman.donald@vydia.test
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <div>
    <div>
      Item indexes to show:
       
      0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19
    </div>
    <div>
      Page:
       
      0
    </div>
    <div>
      Per page:
       
      20
    </div>
    <button
      onClick={[Function]}
    >
      Random page
    </button>
    <button
      onClick={[Function]}
    >
      Change Per Page
    </button>
    <hr />
    <div>
      Item indexes to show:
       
      0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19
    </div>
    <div>
      Page:
       
      0
    </div>
    <div>
      Per page:
       
      20
    </div>
    <button
      onClick={[Function]}
    >
      Random page
    </button>
    <button
      onClick={[Function]}
    >
      Change Per Page
    </button>
  </div>,
]
`;
